device/sycl/src/seeding/counting_grid_capacities.sycl

```diff
@@ -0,0 +1,128 @@
+        auto workItemIdx = item.get_local_id(0);
+
+        // Get device container for input parameters
+        sp_grid_device g2_device(m_grid_view);
+        device_spacepoint_container spacepoints_device(
+            {m_spacepoints_view.headers, m_spacepoints_view.items});
+        vecmem::device_vector<std::pair<unsigned int, unsigned int>>
+            sp_container_indices(m_sp_container_indices_view);
+        vecmem::device_vector<unsigned int> grid_capacities_device(
+            m_grid_capacities_view);
+
+        auto gid = workItemIdx + groupIdx * groupDim;
+
+        /// kill the process before overflow
+        if (gid >= sp_container_indices.size()) {
+            return;
+        }
+
+        const auto& header_idx = sp_container_indices[gid].first;
+        const auto& sp_idx = sp_container_indices[gid].second;
+
+        auto spacepoints_per_module =
+            spacepoints_device.get_items().at(header_idx);
+        const auto& sp = spacepoints_per_module[sp_idx];
+
+        /// Check out if the spacepoints can be used for seeding
+        size_t r_index = is_valid_sp(m_config, sp);
+
+        /// Get axis information from grid
+        const auto& phi_axis = g2_device.axis_p0();
+        const auto& z_axis = g2_device.axis_p1();
+
+        /// Ignore is radius index is invalid value
+        if (r_index != detray::invalid_value<size_t>()) {
+
+            auto isp = internal_spacepoint<spacepoint>(
+                spacepoints_device, {header_idx, sp_idx},
+                vector2{m_config.beamPos[1], m_config.beamPos[0]});
+            /// Get bin index in grid
+            size_t bin_index =
+                phi_axis.bin(isp.phi()) + phi_axis.bins() * z_axis.bin(isp.z());
+
+            /// increase the capacity for the grid bin
+            vecmem::atomic<uint32_t> obj(&grid_capacities_device[bin_index]);
+            obj.fetch_add(1);
+        }
+    }
+
+    private:
+    const seedfinder_config m_config;
+    sp_grid_view m_grid_view;
+    spacepoint_container_view m_spacepoints_view;
+    vecmem::data::vector_view<std::pair<unsigned int, unsigned int>>
+        m_sp_container_indices_view;
+    vecmem::data::vector_view<unsigned int> m_grid_capacities_view;
+};
+
+void counting_grid_capacities(
+    const seedfinder_config config, sp_grid& grid,
+    host_spacepoint_container& spacepoints,
+    vecmem::vector<std::pair<unsigned int, unsigned int>>& sp_container_indices,
+    vecmem::vector<unsigned int>& grid_capacities,
+    vecmem::memory_resource& resource, ::sycl::queue* q) {
+
+    auto grid_view = get_data(grid, resource);
+    auto spacepoints_view = get_data(spacepoints, &resource);
+    auto sp_container_indices_view = vecmem::get_data(sp_container_indices);
+    auto grid_capacities_view = vecmem::get_data(grid_capacities);
+
+    // The local number of threads per work-group (block in cuda) (number taken
+    // from cuda implementation, warp size *2)
+    unsigned int localSize = 32 * 4;
```

Comment:
We will need to find a better way of establishing these values using the information provided by SYCL about the device that we are running on... :thinking: